manifests/extension/config.pp

```diff
@@ -76,10 +76,17 @@
     String => ensure_prefix($settings, "${settings_prefix}."),
   }
 
-  $final_settings = deep_merge(
-    {"${extension_key}" => "${module_path}${so_name}.so"},
-    $full_settings
-  )
+  case $so_name {
```

Comment:
I'm getting a little dizzy trying to decipher this whole logic, including the `$full_settings` bit above.

Could you add some comments to make this more clear?

Reply:
hi @ffrank,

I've had to revisit the code myself to figure out what it was doing...

This define gets a `$settings`-Hash as parameter. If you have it configured to prefix the settings then it will do so resulting in the `$full_settings`-Hash. 

The previous behaviour would always a setting: 
```
extension = /path/to/so_name.so
```

This doesn't work anymore with the MySQL-module in 7.x since the `mysql` extension was removed in favour of `mysqli`, `mysqlnd` or `pdo_mysql`. Which is the second block of code to enable those modules...

If needed I will add more comment tonight in the code.

Reply:
@mcrauwel sorry for the late reply.
Yes, please add some code comments.